modules/ROOT/pages/logger-component-reference.adoc

```diff
@@ -4,76 +4,222 @@ include::_attributes.adoc[]
-`category="MyCustomCategory"`
+Example: `level="ERROR"`
 
+| Category | `category` | Optional string that specifies a category name that it adds to `log4j2.xml` file. For example, you might use a category to route your log messages based on your category, or you might set log levels based on category. Example:
+`category="MyCustomCategory"`.
 |===
 
 == Examples
 
-This Logger is set to monitor message processing status, mixing Strings and expressions:
+The following example you specifies a category `category="MyCustomCategory"` with the `message="My Message"`:
 
-[source,xml,linenums]
+[source,xml]
+----
+<logger level="INFO" doc:name="Logger"
+        message='#["My Message"]' category="MyCustomCategory"/>
+----
+
+Notice that the category `MyCustomCategory` replaces the full package name of the processor,  `org.mule.runtime.core.internal.processor.LoggerMessageProcessor`, for example:
+
+[source,logs]
+----
+INFO  2022-11-11 15:17:53,138 [[MuleRuntime].uber.04:
+[w-logger].logger-example-Flow.CPU_LITE @7ea85211]
+[processor: logger-example-Flow/processors/1;
+event: 113d6700-6217-11ed-86de-147ddaaf4f97]
+MyCustomCategory: My Message
+----
+
+The following example places a DataWeave selector expression within a `message` string:
+
+[source,xml]
 ----
 <logger category="monitoring" message="Message #[payload.id] processed successfully" level="INFO"/>
 ----
 
-This Logger set to record the processing time of a flow, using a single expression and DataWeave's String concatenation:
+Assuming a payload such as `{"id" : "100", "some-key" : "some value" }`, the Logger prints the message `Message 100 processed successfully`, for example:
 
-[source,xml,linenums]
+[source,xml]
+----
+INFO  2022-11-10 21:11:07,791 [[MuleRuntime].uber.09:
+[w-logger].logger-example-Flow.CPU_LITE @600dcddb]
+[processor: logger-example-Flow/processors/1;
+event: 3fd13710-617f-11ed-900f-147ddaaf4f97]
+org.mule.runtime.core.internal.processor.LoggerMessageProcessor:
+Message 100 processed successfully
+----
+
+This Logger records the processing time of a flow, using a single expression and DataWeave concatenation:
+
+//TODO: Need to show how to get processing time or use a better example that doesn't throw an error.
+[source,xml]
 ----
- <logger category="performance" message="#['Message ' ++ payload.id ++ ' took ' ++ vars.processingTime ++ ' milliseconds to process']" level="INFO"/>
+<logger category="performance" message="#['Message ' ++ payload.id ++ ' took ' ++ vars.processingTime ++ ' milliseconds to process']" level="INFO"/>
 ----
 
-== See also
+[[non_repeatable_streams]]
+== Non-Repeatable Streams
+
+In Mule, a non-repeatable stream can be read only once. The Logger consumes a payload that is streamed in a non-repeatable way and does not pass it to the next component in the flow. In the next component, the `payload` selector returns a `null` or empty value. 
```

Comment:
When Logger reads a value that's a non-repeatable stream, that value is consumed. 

Logger itself doesn't return anything (void). 

Transform Message is mutating the payload. Reads and applies a transformation.

Reply:
Output of Logger is void.